mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/utils/GridFsUtil.java

```diff
@@ -0,0 +1,157 @@
+import org.hibernate.ogm.datastore.mongodb.options.impl.BinaryStorageOption;
+import org.hibernate.ogm.datastore.mongodb.options.impl.GridFSBucketOption;
+import org.hibernate.ogm.model.key.spi.EntityKey;
+import org.hibernate.ogm.model.spi.Tuple;
+import org.hibernate.ogm.model.spi.Tuple.SnapshotType;
+import org.hibernate.ogm.options.spi.OptionsContext;
+import org.hibernate.ogm.options.spi.OptionsService;
+
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoCursor;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.gridfs.GridFSBucket;
+import com.mongodb.client.gridfs.GridFSBuckets;
+import org.bson.Document;
+import org.bson.types.ObjectId;
+
+/**
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ */
+public class GridFsUtil {
+	private static final Log log = LoggerFactory.make( MethodHandles.lookup() );
+
+	public static void storeContentToGridFs(MongoDatabase mongoDatabase, Document currentDocument,
+											EntityKey entityKey, OptionsService optionService, Tuple tuple, SnapshotType snapshotType) {
+		Class entityClass = TableEntityTypeMappingInfo.getEntityClass( entityKey.getTable() );
+		if ( currentDocument == null ) {
+			return;
+		}
+		for ( String fieldName : currentDocument.keySet() ) {
+			if ( fieldName.equals( "$set" ) ) {
+				// it is part of request. it is not document
+				Document queryFields = (Document) currentDocument.get( fieldName );
+				for ( String queryField : queryFields.keySet() ) {
+					loadContent( mongoDatabase,queryFields,entityClass,queryField,optionService,tuple,snapshotType );
+				}
+			}
+			else {
+				// it is not document
+				loadContent( mongoDatabase,currentDocument,entityClass,fieldName,optionService,tuple,snapshotType );
+			}
+		}
+	}
+
+	private static void loadContent(MongoDatabase mongoDatabase,  Document currentDocument, Class entityClass, String fieldName,
+									OptionsService optionService,Tuple tuple, SnapshotType snapshotType) {
+		OptionsContext optionsContext = getPropertyOptions( optionService, entityClass, fieldName );
+		BinaryStorageType binaryStorageType = optionsContext.getUnique( BinaryStorageOption.class );
+		if ( BinaryStorageType.GRID_FS == binaryStorageType  ) {
+			//the field has GridFS configuration. Process it!
+			String gridfsBucketName = optionsContext.getUnique( GridFSBucketOption.class );
+
+			GridFSBucket gridFSFilesBucket = getGridFSFilesBucket( mongoDatabase, gridfsBucketName );
+			String fileName = "";
+			BinaryStream binaryStream = currentDocument.get( fieldName,BinaryStream.class );
+			ObjectId uploadId = gridFSFilesBucket.uploadFromStream( fileName, binaryStream.getInputStream() );
+			//change value of the field (BinaryStream -> ObjectId)
+			currentDocument.put( fieldName, uploadId );
+			if ( SnapshotType.UPDATE == snapshotType ) {
+				ObjectId oldContentObjectId = (ObjectId) tuple.get( fieldName + "_uploadId" );
+				gridFSFilesBucket.delete( oldContentObjectId );
+			}
```

Comment:
I think the methods in this class do too many things. I would expect them to be called when you already know that you need the GridFS approach. Also, I wouldn't add references to snapshotType in it.

Reply:
HI @DavideD !
I will create new file in GridFS because binary content can be changed. In this way, I must remove previous upload to GridFS.
If you not agree with using snapshotType  I can check field "_uploadId". And I will remove previous upload to GridFS if the field exists in document.

Is it ok ?